ios/BT/ExposureManager.swift

```diff
@@ -193,19 +183,59 @@ final class ExposureManager: NSObject {
                 if !newExposures.isEmpty {
                   self.postNewExposureNotification()
                 }
-                finish(.success(newExposures))
+                self.finish(.success(newExposures),
+                            processedFileCount: processedFileCount,
+                            lastProcessedUrlPath: lastProcessedUrlPath,
+                            progress: progress,
+                            completionHandler: completionHandler)
               }
             }
           }
         } catch(let error) {
-          finish(.failure(error))
+          self.finish(.failure(error),
+                      processedFileCount: processedFileCount,
+                      lastProcessedUrlPath: lastProcessedUrlPath,
+                      progress: progress,
+                      completionHandler: completionHandler)
         }
       }
     }
-    
     return progress
   }
   
+  func finish(_ result: Result<[Exposure]>,
+              processedFileCount: Int,
+              lastProcessedUrlPath: String,
+              progress: Progress,
+              completionHandler: ((Error?) -> Void)) {
+    
+    cleanup()
+    
+    if progress.isCancelled {
+      detectingExposures = false
+      postExposureDetectionErrorNotification()
+      BTSecureStorage.shared.exposureDetectionErrorLocalizedDescription = GenericError.unknown.localizedDescription
+      completionHandler(GenericError.unknown)
+    } else {
+      switch result {
+      case let .success(newExposures):
+        BTSecureStorage.shared.exposureDetectionErrorLocalizedDescription = .default
+        BTSecureStorage.shared.remainingDailyFileProcessingCapacity -= processedFileCount
+        if lastProcessedUrlPath != .default {
```

Comment:
If I'm understanding this correctly we are updating the URL unless the path is the default one. Are there ramifications to update it to the default even if it hasn't changed? I feel like it would save us this if and maybe drop complexity a tiny bit? Definitely not a big deal just curious.

Reply:
`.default` is an extension on `String` that compiles to the empty string (`""`)